main.py

```diff
@@ -44,15 +59,20 @@ def apply_custom_paths():
                                        os.path.join(folder_paths.get_output_directory(), "diffusion_models"))
     folder_paths.add_model_folder_path("loras", os.path.join(folder_paths.get_output_directory(), "loras"))
 
-    if args.input_directory:
-        input_dir = os.path.abspath(args.input_directory)
-        logging.info(f"Setting input directory to: {input_dir}")
-        folder_paths.set_input_directory(input_dir)
 
-    if args.user_directory:
-        user_dir = os.path.abspath(args.user_directory)
-        logging.info(f"Setting user directory to: {user_dir}")
-        folder_paths.set_user_directory(user_dir)
+# These are created at startup if missing. Other data directories are created on-demand when needed.
+def create_data_folders():
+    data_dirs = [
+        folder_paths.get_input_directory(),
+        folder_paths.get_temp_directory()
+    ]
```

Comment:
Optionally create the `user/` directory here (_`UserManager` already does so at startup?_), and restore the `output/` directory creation (_presumably not necessary and is instead handled on-demand_).

```suggestion
    data_dirs = [
        folder_paths.get_input_directory(),
        folder_paths.get_output_directory(),
        folder_paths.get_temp_directory(),
        folder_paths.get_user_directory()
    ]
```